raku-advent-2020/local-drafts/p6steve-rudolph-on-raku.md

```diff
@@ -0,0 +1,183 @@
+# Rudolph on Raku
+
+## Finding a way home to the North pole with Physics::Navigation
+
+So, Rudolph has been worried about getting Santa and the other reindeer back home to the North Pole 
+after an exhausting flight to visit all the (well-behaved) Children on the Globe.
+
+He has heard a rumour that the North Pole keeps moving due to the precession of molten iron at 
+the Earth's core and that every year it creeps around a bit with relation to Santa's workshop,
+which lies at the True North Pole.
+
+Luckily he has been on a navigation skills course and has learned about how to specify a position on
+the globe using a combination of Latitude and Longitude. However, these seem all of a muddle as they 
+are alike and yet different. What Rudi needs is a way to structure his navigation to ensure that he 
+does not mix them up. Even better, he is good friends with Larry and knows that he can trust the 
+Raku type system to get him home.
+
+Let's see how he does it:
+
+```
+use Physics::Unit;
+use Physics::Measure;
+
+class NavAngle is Angle {
+  has Unit  $.units is rw where *.name eq '°';
+	
+  multi method new( Str:D $s ) {
+    my ($decimal, $compass) = NavAngle.defn-extract( $s );
+    my $type;
+    given $compass {
+      when <N S>.any   { $type = 'Latitude' }
+      when <E W>.any   { $type = 'Longitude' }
+      when <M T H>.any { $type = 'Bearing' }
+      default          { nextsame }
+    }
+    ::($type).new( value => $nominal, compass => $compass );
+  }
+  
+  method defn-extract( NavAngle:U: Str:D $s ) {
+    #handle degrees-minutes-seconds <°> is U+00B0 <′> is U+2032 <″> is U+2033
+    unless $s ~~ /(\d*)\°(\d*)\′?(\d*)\″?\w*(<[NSEWMTH]>)/ { return 0 };
+    my $deg where 0 <= * < 360 = $0 % 360;
+    my $min where 0 <= * <  60 = $1 // 0;
+    my $sec where 0 <= * <  60 = $2 // 0;
+    my $decimal = ( ($deg * 3600) + ($min * 60) + $sec ) / 3600;
+    my $compass = ~$3;
+
+    say "NA extracting «$s»: value is $deg°$min′$sec″, compass is $compass" if $db;
+    return($decimal, $compass)
+  }
+  
+  method Str {
+    my ( $deg, $min ) = self.dms( :no-secs ); 
+    $deg = sprintf( "%03d", $deg );
+    qq{$deg° $.compass}
+  }
+}
+#real code at https://github.com/p6steve/raku-Physics-Navigation (work in progress)
+```
+So Rudi has inherited the Angle type provided by Physics::Unit and created some general methods that 'know' that 
```

Comment:
Please try and provide a extended explanation. For instance, say that "is" is used for inheritance, even saying prior to the example how to download the module. Don't be afraid of being verbose; we want the articles to be as self-contained and beginner-friendly as possible.

Reply:
done - good feedback